gaskij/rpicampusmap
# Create the Info Page in React

## Description
The info page for each location had not yet been brought over from the AngularJs to React refactor. With this PR, the Info page now exists and is linked to from the map and search results pages. You can also visit a location directly from the URL.

Fixes #142 

## Solution
* Created the Info Page in React
* Changed the link to the page to use a nested url instead of a parameter string (`/info?loc=dcc` --> `/info/dcc`)
* Changed the webpack configuration to properly allow nested URLs and still serve the JS file from the correct public url
* Fleshed out the `PhotoCarousel` and `Comment` components

## Known Issues
* The info page may be overhauled at some point in the future, but it is important to have a basic implementation first
* The public image upload functionality has been scrapped, but may return in the future
* The comments and photos are using fake data, since the real stuff is not there yet. The backend API endpoints do not yet exist for getting photos and comments
* Comment submission does not yet work, as the API endpoints do not yet exist.
* The photos may be different sizes and cause the layout to shift when the photo carousel changes. When the actual photos are used, this shouldn't be a problem, since they will be the same size.
* _**!!This PR includes changes from PR #146. #146 should be reviewed and merged before this one!!**_
* _**!!This PR includes a basic Not Found page implementation. This will be improved in a future iteration and should not be judged for quality!!**_

## Testing Procedures
* Navigate to the info page from the map page by clicking a location, then clicking its popup. Ensure it is the correct location and the page loads properly.
* Navigate to the info page from the search results page by clicking a location. Ensure it is the correct location and the page loads properly.
* Navigate to the info page from directly through the browser at `/info/:id`. Ensure it is the correct location and the page loads properly.
* Play around with the photo carousel and ensure the photos change.
* Click "Show on Map" and ensure it focuses on the correct location.

## Checklist for author
